app/models/github_repository.rb

```diff
@@ -13,6 +13,10 @@ def get_starter_code_from(source)
     end
   end
 
+  def present?
```

Comment:
@cyhsutw I'd like to add an optional `options` hash to these methods.

```ruby
def present?(**options)
  self.class.present?(@client, @id)
end

def self.present?(client, repo_identifier, **options)
  client.repository?(full_name, options)
end
```

This way when we ask if the repo is present we can do it without caching.

```ruby
# Example
assgnment_repo.github_repository.present?(GitHub::APIHeaders.no_cache_no_store)
```